File: code36.py
class Solution(object):
    def isValidSudoku(self, board):
        check=set()
        print(board[3])
        print(board[:,4])
        print(board[3:7,3:7])
        for r in range(0,len(board)):
            for c in range(0,len(board[r])):
                v=board[r][c]
                if v=='.':
                    continue
                key_row='r'+str(r)+'v:'+v
                key_col='c'+str(r)+'v:'+v
                key_box='b'+str(r//3)+str(c//3)+'v:'+v
                print(key_row,key_col,key_box)
                if key_row in check or key_col in check or key_box in check:
                    return False
                check.add(key_box)
                check.add(key_row)
                check.add(key_col)
        return True
s=Solution()
brd=[["5","3",".",".","7",".",".",".","."],["6",".",".","1","9","5",".",".","."],[".","9","8",".",".",".",".","6","."],["8",".",".",".","6",".",".",".","3"],["4",".",".","8",".","3",".",".","1"],["7",".",".",".","2",".",".",".","6"],[".","6",".",".",".",".","2","8","."],[".",".",".","4","1","9",".",".","5"],[".",".",".",".","8",".",".","7","9"]]
brd2=[["8","3",".",".","7",".",".",".","."],["6",".",".","1","9","5",".",".","."],[".","9","8",".",".",".",".","6","."],["8",".",".",".","6",".",".",".","3"],["4",".",".","8",".","3",".",".","1"],["7",".",".",".","2",".",".",".","6"],[".","6",".",".",".",".","2","8","."],[".",".",".","4","1","9",".",".","5"],[".",".",".",".","8",".",".","7","9"]]
brd3=[[".",".","4",".",".",".","6","3","."],[".",".",".",".",".",".",".",".","."],["5",".",".",".",".",".",".","9","."],[".",".",".","5","6",".",".",".","."],["4",".","3",".",".",".",".",".","1"],[".",".",".","7",".",".",".",".","."],[".",".",".","5",".",".",".",".","."],[".",".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".",".","."]]
print(s.isValidSudoku(brd))
print(s.isValidSudoku(brd2))
print(s.isValidSudoku(brd3))

File: code36.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    bool isValidSudoku(vector<vector<char>> &board)
    {

        for (int i = 0; i < 9; i++)
        {
            unordered_map<char, bool> seen1;
            unordered_map<char, bool> seen2;
            for (int j = 0; j < 9; j++)
            {
                char c1 = board[i][j];

                char c2 = board[j][i];
                if (seen1[c1] && c1 != '.')
                    return false;
                if (seen2[c2] && c2 != '.')
                    return false;
                seen1[c1] = true;
                seen2[c2] = true;
            }
        }

        for (int i = 0; i < 9; i += 3)
        {
            for (int j = 0; j < 9; j += 3)
            {
                unordered_map<char, bool> seen;

                for (int x = i; x < i + 3; x++)
                {
                    for (int y = j; y < j + 3; y++)
                    {
                        char c = board[x][y];
                        if (seen[c] && c != '.')
                            return false;
                        seen[c] = true;
                    }
                }
            }
        }
        return true;
    }
};